### ISSUE TITLE
Use tiles with rounded zoom levels

### ISSUE THREAD

**Issue by Maradox** [osmdroid/osmdroid#1824]
When zooming the tiles are scaled with the tiles of the lower zoom-level. This causes the tiles to be pixelated.
Using the tiles of the upper zoom-level should result in a better resolution.

An example: We zoom from 4.0 to 5.0
Until 4.99 the tiles from zoom-level 4 are used, which causes the drawn tiles between 4 and 5 to be very pixelated.
Once we reach zoom 5.0 the tiles are sharp again since the tiles from zoom-level 5 are used.

Is there a way to modify this behavior for example by using the tiles of the rounded zoom level?

**Comment by monsieurtanuki** [comment]
Hi @Maradox!
I would try `mapView.setZoomRounding(true);` (cf. #944).
Please close the issue if it works.

**Comment by Maradox** [comment]
The method `mapView.setZoomRounding(true)` snaps the zoom. 

I still would like to have continuous zoom, but when scaling tiles the tiles with a higher zoom-level should be used.

**Comment by Maradox** [comment]
You are correct, my idea was that four zoom-level-5 tiles are used starting at zoom >= 4.01. 
I'm not sure why for example at zoom 4.99 - four zoom-level-5 tiles would not have a better resolution than one zoom-level-4 tile?

**Comment by monsieurtanuki** [comment]
Assuming that the tile png's are correctly drawn, if you downsize them you see less things, texts are less readable and you may not see roads or rivers anymore.
That said, the impact on level 4.99 would be limited.

**Comment by eclectice** [comment]
Just to note, in every GIS tool, constructing the tile pyramid of a map file is always a zoom list in round numbers...
